cpp/src/join/conditional_join.cu

```diff
@@ -50,8 +145,8 @@ conditional_join(table_view const& left,
   // We can immediately filter out cases where the right table is empty. In
   // some cases, we return all the rows of the left table with a corresponding
   // null index for the right table; in others, we return an empty output.
-  auto right_num_rows{right.num_rows()};
-  auto left_num_rows{left.num_rows()};
+  auto const right_num_rows{right.num_rows()};
+  auto const left_num_rows{left.num_rows()};
```

Comment:
Let's choose a consistent pattern between this and the changes below. Either use `auto const` variables or always use `.num_rows()`. It looks like the below was refactored to use the method, but this is still using variables.